PennMobile/src/main/java/com/pennapps/labs/pennmobile/gsrFragment.java

```diff
@@ -0,0 +1,593 @@
+                        }, mYear, mMonth, mDay);
+
+                //set min and max choices for dates. Want to limit to week.
+                Date today = new Date();
+                c.setTime(today);
+                long minDate = c.getTime().getTime();
+
+                c.setTime(today);
+                c.add( Calendar.DAY_OF_MONTH, +6 );
+                long maxDate = c.getTime().getTime();
+
+                datePickerDialog.getDatePicker().setMaxDate(maxDate);
+                datePickerDialog.getDatePicker().setMinDate(minDate);
+                datePickerDialog.show();
+            }
+
+        });
+
+        //execute the seatch
+        searchGSR.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                Toast.makeText(getActivity(), "Loading...", Toast.LENGTH_SHORT).show();
+
+                instructions.setText(getString(R.string.select_intrusctions));
+
+                //get vars
+                String dateBooking = calendarButton.getText().toString();
+                String startTime = startButton.getText().toString();
+                String endTime = endButton.getText().toString();
+                int location = mapGSR(gsrDropDown.getSelectedItem().toString());
+                if (location == -1) {
+                    Toast.makeText(getActivity(), "Sorry, an error has occurred", Toast.LENGTH_LONG).show();
+                }
+                else {
+                    //get the hours
+                    String[] asyncTaskParams = { Integer.toString(location), dateBooking, startTime, endTime};
+                    new getHours().execute(asyncTaskParams);
+                }
+
+            }
+
+        });
+
+        /**
+         *
+         *
+         * END on click functions for buttons
+         *
+         */
+
+        //load initial data
+        loadInitialData();
+
+        return v;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        getActivity().setTitle(R.string.gsr);
+        ((MainActivity) getActivity()).setNav(R.id.nav_gsr);
+    }
+
+    //takes the name of the gsr and returns an int for the corresponding code
+    public int mapGSR(String name) {
+        switch (name) {
+            case "Weigle":
+                return 1722;
+            case "VP GSR":
+                return 1799;
+            case "Lippincott":
+                return 1768;
+            case "Edu Commons":
+                return 848;
+            case "Levin Building":
+                return 13489;
+            case "VP Sem. Rooms":
+                return 4409;
+            case "Lippincott Sem. Rooms":
+                return 2587;
+            case "Glossberg Recording Room":
+                return 1819;
+            case "Dental Sem":
+                return 13532;
+            case "Biomedical Lib.":
+                return 505;
+            default:
+                return -1;
+        }
```

Comment:
Maybe use a `HashMap` instead of this? Just a thought, not sure how feasible in Android Framework Java